web/source/osk/oskLayerGroup.ts

```diff
@@ -0,0 +1,94 @@
+/// <reference path="oskLayer.ts" />
+
+namespace com.keyman.osk {
+  export class OSKLayerGroup {
+    public readonly element: HTMLDivElement;
+    public readonly layers: {[layerID: string]: OSKLayer} = {};
+
+    public constructor(vkbd: VisualKeyboard, keyboard: keyboards.Keyboard, formFactor: utils.FormFactor) {
+      let layout = keyboard.layout(formFactor);
+      let oskManager = com.keyman.singleton.osk;
+
+      const lDiv = this.element = document.createElement('div');
+      const ls=lDiv.style;
+
+      // Set OSK box default style
+      lDiv.className='kmw-key-layer-group';
+
+      // Return empty DIV if no layout defined
+      if(layout == null) {
+        return;
+      }
+
+      // Set default OSK font size (Build 344, KMEW-90)
+      let layoutFS = layout['fontsize'];
+      if(typeof layoutFS == 'undefined' || layoutFS == null || layoutFS == '') {
+        ls.fontSize='1em';
+      } else {
+        ls.fontSize=layout['fontsize'];
+      }
+
+      vkbd.fontSize=ls.fontSize;       //TODO: move outside function*********
+
+      // Create a separate OSK div for each OSK layer, only one of which will ever be visible
+      var n: number, i: number, j: number;
+      var layers: keyboards.LayoutLayer[];
+
+      layers=layout['layer'];
+
+      // Set key default attributes (must use exportable names!)
+      var tKey=vkbd.getDefaultKeyObject();
+      tKey['fontsize']=ls.fontSize;
+
+      // ***Delete any empty rows at the end added by compiler bug...
+      for(n=0; n<layers.length; n++) {
+        let layer=layers[n];
+        let rows=layer['row'];
+        for(i=rows.length; i>0; i--) {
+          if(rows[i-1]['key'].length > 0) {
+            break;
+          }
+        }
+
+        if(i < rows.length) {
+          rows.splice(i-rows.length,rows.length-i);
+        }
+      }
+      // ...remove to here when compiler bug fixed ***
```

Comment:
Figured as much.  That's why I've always made sure to preserve it.

Reply:
Good to hear! So probably we should delete the comment 👍

Reply:
... you know, now that I think of it... this bug-fix should probably be part of Web-Core, rather than part of layer-group construction.  Should anyone else ever want to use Web's `KeyboardProcessor` directly, they'd need this too.

Reply:
I've gone ahead and added a commit dedicated to that last comment:  https://github.com/keymanapp/keyman/pull/5451/commits/ed490fcbeffa12ecbd9dc7fdd7a42e35e5d6c720